cherab/tools/observers/spectroscopic.py

```diff
@@ -101,105 +428,222 @@ def plot_spectra(self, unit='J', ymax=None):
+            if len(value) == len(self._sight_lines):
+                for sight_line, v in zip(self._sight_lines, value):
+                    sight_line.acceptance_angle = v
+            else:
+                raise ValueError("The length of 'acceptance_angle' ({}) "
+                                 "mismatches the number of sight-lines ({}).".format(len(value), len(self._sight_lines)))
+        else:
+            for sight_line in self._sight_lines:
+                sight_line.acceptance_angle = value
+
+    @property
+    def radius(self):
+        """
+        The radius of the fibre tip in metres. This radius defines a circular area at the fibre tip
+        which will be sampled over.
+
+        The same value can be shared between all sight lines,
+        or each sight line can be assigned with individual value.
+
+        :rtype: list
+        """
+        return [sight_line.radius for sight_line in self._sight_lines]
+
+    @radius.setter
+    def radius(self, value):
+        if isinstance(value, (list, tuple, ndarray)):
+            if len(value) == len(self._sight_lines):
+                for sight_line, v in zip(self._sight_lines, value):
+                    sight_line.radius = v
+            else:
+                raise ValueError("The length of 'radius' ({}) "
+                                 "mismatches the number of sight-lines ({}).".format(len(value), len(self._sight_lines)))
+        else:
+            for sight_line in self._sight_lines:
+                sight_line.radius = value
+
+
+class _SpectroscopicObserver0DBase:
+    """A base class for spectroscopic 0D observers."""
 
     @property
     def point(self):
+        """
+        The observation point of the sight line.
+
+        :rtype: Point3D
+        """
         return self._point
 
     @point.setter
     def point(self, value):
-        if not (self._direction.x == 0 and self._direction.y == 0 and self._direction.z == 1):
+        if not isinstance(value, Point3D):
+            raise TypeError("Attribute 'point' must be of type Point3D.")
+
+        if self._direction.x != 0 or self._direction.y != 0 or self._direction.z != 1:
             up = Vector3D(0, 0, 1)
         else:
             up = Vector3D(1, 0, 0)
         self._point = value
-        self._observer.transform = translate(value.x, value.y, value.z) * rotate_basis(self._direction, up)
+        self.transform = translate(value.x, value.y, value.z) * rotate_basis(self._direction, up)
 
     @property
     def direction(self):
+        """
+        The observation direction of the sight line.
+
+        :rtype: Vector3D
+        """
         return self._direction
 
     @direction.setter
     def direction(self, value):
-        if value.x != 0 and value.y != 0 and value.z != 1:
+        if not isinstance(value, Vector3D):
+            raise TypeError("Attribute 'direction' must be of type Vector3D.")
+
+        if value.x != 0 or value.y != 0 or value.z != 1:
```

Comment:
Yes, we will probably have to add `orientation` property along with `direction` when polarisation is implemented in Raysect. Since we agreed that @CnlPepper will remove unpolarised rays, we will need to update the existing classes rather than create separate ones. However, until then I would leave it as it is.